src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xencoder/HaivisionXEncoderCommunicator.java

```diff
@@ -0,0 +1,813 @@
+		switch (encoderMonitoringMetric) {
+			case AUDIO_STATISTICS:
+				populateAudioStatisticsData(stats);
+				break;
+			case VIDEO_STATISTICS:
+				populateVideoStatisticsData(stats);
+				break;
+			case STREAM_STATISTICS:
+				populateStreamStatisticsData(stats);
+				break;
+			case AUDIO_CONFIG:
+				populateAudioConfigData(stats, advancedControllableProperties);
+				break;
+			case TEMPERATURE:
+			case ACCOUNT:
+			case SYSTEM_INFORMATION:
+			case ROLE_BASED:
+				break;
+			default:
+				if (logger.isDebugEnabled()) {
+					logger.debug("The metric not support populate data" + encoderMonitoringMetric.getName());
+				}
+		}
+	}
+
+	/**
+	 * Populate audio config
+	 *
+	 * @param stats list statistics property
+	 * @param advancedControllableProperties the advancedControllableProperties is list AdvancedControllableProperties
+	 */
+	private void populateAudioConfigData(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties) {
+		Objects.requireNonNull(stats);
+		Objects.requireNonNull(advancedControllableProperties);
+		//TODO
+	}
+
+	/**
+	 * Populate stream statistics
+	 *
+	 * @param stats list statistics property
+	 */
+	private void populateStreamStatisticsData(Map<String, String> stats) {
+		for (StreamStatistics streamStatistics : streamStatisticsList) {
+			String streamName = streamStatistics.getName();
+			if (EncoderConstant.NONE_STREAM_NAME.equals(streamName)) {
+				streamName = handleStreamNameIsEmpty(streamStatistics.getId());
+			}
+			String metricName = EncoderConstant.STREAM + EncoderConstant.SPACE + streamName + EncoderConstant.SPACE + EncoderConstant.STATISTICS + EncoderConstant.HASH;
+			for (StreamMonitoringMetric streamMonitoringMetric : StreamMonitoringMetric.values()) {
+				String value = getDefaultValueForNullData(streamStatistics.getValueByMetric(streamMonitoringMetric));
+				if (StreamMonitoringMetric.UPTIME.equals(streamMonitoringMetric)) {
+					stats.put(metricName + streamMonitoringMetric.getName(), formatTimeData(value));
+					continue;
+				}
+				stats.put(metricName + streamMonitoringMetric.getName(), replaceCommaByEmptyString(value));
+			}
+		}
+	}
+
+	/**
+	 * Populate audio statistics
+	 *
+	 * @param stats list statistics property
+	 */
+	private void populateAudioStatisticsData(Map<String, String> stats) {
+		for (AudioStatistics audioStatistics : audioStatisticsList) {
+			String audioName = audioStatistics.getName();
+			String metricName = audioName + EncoderConstant.SPACE + EncoderConstant.STATISTICS + EncoderConstant.HASH;
+			for (AudioMonitoringMetric audioMetric : AudioMonitoringMetric.values()) {
+				String value = getDefaultValueForNullData(audioStatistics.getValueByMetric(audioMetric));
+				if (audioMetric.equals(AudioMonitoringMetric.ENCODED_BYTES) || audioMetric.equals(AudioMonitoringMetric.ENCODED_FRAMES)) {
+					stats.put(metricName + audioMetric.getName(), replaceCommaByEmptyString(value));
```

Comment:
metricName + audioMetric.getName() could be extracted into a variable before the if/else block